File: leetcode/C++/remove-covered-intervals.cpp
bool compare(vector<int> &a, vector<int> &b){
    if(a[0]==b[0]) return a[1]>b[1];
    return a[0]<b[0];
}
class Solution {
public:
    int removeCoveredIntervals(vector<vector<int>>& intervals) {
        int sol=0, mx=-1;
        sort(intervals.begin(), intervals.end(), compare);
        for(vector<int> &v : intervals){
            if(mx<v[1]) ++sol;
            mx=max(mx,v[1]);
        }
        return sol;
    }
};

File: leetcode/Python/remove-covered-intervals.py
def compare(a, b):
    if a[0] == b[0]:
        return -1 if a[1]>b[1] else 1
    return -1 if a[0]<b[0] else 1
class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        sol, mx = 0, -1
        intervals.sort(key=cmp_to_key(compare))
        for v in intervals:
            if mx<v[1]:
                sol += 1
            mx = max(mx, v[1])
        return sol